pkg/cmd/admin/registry/registry.go

```diff
@@ -402,6 +405,14 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 		}
 	}
 
+	// if we're using a daemon set we'll add it here and remove the DC.
+	// We allow the dc to be created above to get the goodness of app.AddServices
```

Comment:
That doesn't look too bad to fix.  Why wouldn't we make the update in that function?